### DISCUSSION TITLE
Memory Game WASM Failure

### DISCUSSION THREAD

**Discussion by goldhaxx** [slint-ui/slint#4562] [General]
Hello Slint team! I enjoyed going through the memory game tutorial and as a last step wanted to attempt exporting the project to wasm and viewing it from a local python webserver as shown at the end of your tutorial.

I was able to export everything to wasm by following your instructions, however I'm having issues rendering the page in my browser. 

Running ```repos\memory\pkg> python -m http.server``` from my terminal starts the python webserver within the pkg folder.

```Serving HTTP on :: port 8000 (http://[::]:8000/) ...```

However, when navigating to http://localhost:8000 I am seeing the following error in my terminal:

```::1 - - [06/Feb/2024 22:23:00] "GET /index.html HTTP/1.1" 200 -```
```::1 - - [06/Feb/2024 22:23:00] code 404, message File not found```

Here are the contents of the index.html folder which I placed in the ./pkg directory of the project, along with the memory.js and other js files.

```
<html>
    <body>
        <!-- canvas required by the Slint runtime -->
        <canvas id="canvas"></canvas>
        <script type="module">
            // import the generated file.
            import init from "memory.js";
            init();
        </script>
    </body>
</html>
```

Prior to placing the index.html file in this directory, I also attempted placing it in the project root directory and running the web server from there.

I understand this may be an issue with my webserver config, but if you have any other recommended approaches aside from this I am open to hearing them.

Any pointers here would be greatly appreciated.

**Comment by ogoffart** [comment]
Do your browser try to open memory.js and that is a 404?  Are you sure that this file exist in the pkg directory?
Do you have output after that? if the 404 is for example, for the favicon.ico, or other metadata, you can safely ignore. But maybe the warning comes after.
You can also look at the javascript console from your browser (Ctrl+shift+I, usually)
